src/Ng/CommandHelpers.cs

```diff
@@ -45,275 +55,269 @@ private static void TraceRequiredArgument(string name)
+                CloudStorageAccount account = new CloudStorageAccount(credentials, true);
+                return new AzureStorageFactory(account, storageContainer, storagePath, storageBaseAddress) { Verbose = verbose, CompressContent = true };
             }
-
-            string storageType;
-            if (!arguments.TryGetValue("-storageType", out storageType))
+            catch (ArgumentException)
             {
-                TraceRequiredArgument("-storageType");
                 return null;
             }
+        }
 
-            if (storageType.Equals("File", StringComparison.InvariantCultureIgnoreCase))
+        public static StorageFactory CreateStorageFactory(IDictionary<string, string> arguments, bool verbose)
+        {
+            try
             {
-                string storagePath;
-                if (!arguments.TryGetValue("-storagePath", out storagePath))
+                Uri storageBaseAddress = null;
+                string storageBaseAddressStr;
+                if (!TryGetArgument(arguments, Constants.StorageBaseAddress, out storageBaseAddressStr))
                 {
-                    TraceRequiredArgument("-storagePath");
-                    return null;
-                }
+                    storageBaseAddressStr = storageBaseAddressStr.TrimEnd('/') + "/";
 
-                if (storageBaseAddress == null)
-                {
-                    TraceRequiredArgument("-storageBaseAddress");
-                    return null;
+                    storageBaseAddress = new Uri(storageBaseAddressStr);
                 }
 
-                return new FileStorageFactory(storageBaseAddress, storagePath) { Verbose = verbose };
-            }
-            else if (storageType.Equals("Azure", StringComparison.InvariantCultureIgnoreCase))
-            {
-                string storageAccountName;
-                if (!arguments.TryGetValue("-storageAccountName", out storageAccountName))
-                {
-                    TraceRequiredArgument("-storageAccountName");
-                    return null;
-                }
+                string storageType;
+                TryGetArgument(arguments, Constants.StorageType, out storageType, required: true);
 
-                string storageKeyValue;
-                if (!arguments.TryGetValue("-storageKeyValue", out storageKeyValue))
+                if (storageType.Equals("File", StringComparison.InvariantCultureIgnoreCase))
                 {
-                    TraceRequiredArgument("-storageKeyValue");
-                    return null;
+                    string storagePath;
+                    TryGetArgument(arguments, Constants.StoragePath, out storagePath, required: true);
+
+                    if (storageBaseAddress == null)
+                    {
+                        TraceRequiredArgument("-storageBaseAddress");
+                        return null;
+                    }
```

Comment:
storageBaseAddress is not the var checked directly above. it is reused and required for File type, but not for Azure type. So we null check here.

---
In reply to: [71207829](https://github.com/NuGet/NuGet.Services.Metadata/pull/77#discussion_r71207829) [](ancestors = 71207829)